Xamarin.Forms.Core/BindableObject.cs

```diff
@@ -486,7 +486,7 @@ BindablePropertyContext GetContext(BindableProperty property)
 			for (var i = 0; i < properties.Count; i++)
 			{
 				BindablePropertyContext context = properties[i];
-				if (ReferenceEquals(context.Property, property))
+				if (context != null && ReferenceEquals(context.Property, property))
```

Comment:
@dnfclas Is not always. Often errors were in second thread but also errors were in main thread. 

I had these crashes:

```
Xamarin caused by: android.runtime.JavaProxyThrowable: System.NullReferenceException: Object reference not set to an instance of an object.
  at Xamarin.Forms.BindableObject.GetContext (Xamarin.Forms.BindableProperty property) [0x00013] in C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Core\BindableObject.cs:489 
  at Xamarin.Forms.BindableObject.GetValue (Xamarin.Forms.BindableProperty property) [0x0000e] in C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Core\BindableObject.cs:54 
  at Xamarin.Forms.TemplatedView.get_ControlTemplate () [0x00000] in C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Core\TemplatedView.cs:13 
  at Xamarin.Forms.TemplatedView.SetChildInheritedBindingContext (Xamarin.Forms.Element child, System.Object context) [0x00000] in C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Core\TemplatedView.cs:64 
  at Xamarin.Forms.Element.set_Parent (Xamarin.Forms.Element value) [0x0007b] in C:\BuildAgent2\work\aad494dc9bc9783\Xamarin.Forms.Core\Element.cs:196 
  at Rg.Forms.ThreadView.Droid.Renderers.Controls.ThreadViewRenderer+<>c__DisplayClass3_0+<<CreateRenderer>b__0>d.MoveNext () [0x000e4] in <c34ebe9604734dabb7434028866eb64f>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () [0x0000c] in /Users/builder/data/lanes/3511/f4db8a57/source/mono/mcs/class/referencesource/mscorlib/system/runtime/exceptionservices/exceptionservicescommon.cs:143 
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore.<ThrowAsync>m__1 (System.Object state) [0x00000] in /Users/builder/data/lanes/3511/f4db8a57/source/mono/mcs/class/referencesource/mscorlib/system/runtime/compilerservices/AsyncMethodBuilder.cs:1034 
  at System.Threading.QueueUserWorkItemCallback.WaitCallback_Context (System.Object state) [0x0000e] in /Users/builder/data/lanes/3511/f4db8a57/source/mono/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1304 
  at System.Threading.ExecutionContext.RunInternal (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x0008d] in /Users/builder/data/lanes/3511/f4db8a57/source/mono/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:957 
  at System.Threading.ExecutionContext.Run (System.Threading.ExecutionContext executionContext, System.Threading.ContextCallback callback, System.Object state, System.Boolean preserveSyncCtx) [0x00000] in /Users/builder/data/lanes/3511/f4db8a57/source/mono/mcs/class/referencesource/mscorlib/system/threading/executioncontext.cs:904 
  at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem () [0x0002a] in /Users/builder/data/lanes/3511/f4db8a57/source/mono/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1281 
  at System.Threading.ThreadPoolWorkQueue.Dispatch () [0x00096] in /Users/builder/data/lanes/3511/f4db8a57/source/mono/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:854 
  at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback () [0x00000] in /Users/builder/data/lanes/3511/f4db8a57/source/mono/mcs/class/referencesource/mscorlib/system/threading/threadpool.cs:1209 
```

```
Xamarin caused by: android.runtime.JavaProxyThrowable: System.NullReferenceException: Object reference not set to an instance of an object
  at Xamarin.Forms.BindableObject.GetContext (Xamarin.Forms.BindableProperty property) [0x00013] in <b0fc14d4e5b04749b7241d1235a68329>:0 
  at Xamarin.Forms.BindableObject.GetValue (Xamarin.Forms.BindableProperty property) [0x00016] in <b0fc14d4e5b04749b7241d1235a68329>:0 
  at Xamarin.Forms.VisualElement.get_Width () [0x00018] in <b0fc14d4e5b04749b7241d1235a68329>:0 
  at Xamarin.Forms.Platform.Android.VisualElementTracker.UpdateLayout () [0x0003e] in <95895bcb439e42ebb8e7134a545f1acc>:0 
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].UpdateLayout () [0x0000e] in <95895bcb439e42ebb8e7134a545f1acc>:0 
  at Xamarin.Forms.Platform.Android.VisualElementRenderer`1[TElement].OnLayout (System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x0003d] in <95895bcb439e42ebb8e7134a545f1acc>:0 
  at Xamarin.Forms.Platform.Android.ViewRenderer`2[TView,TNativeView].OnLayout (System.Boolean changed, System.Int32 l, System.Int32 t, System.Int32 r, System.Int32 b) [0x00000] in <95895bcb439e42ebb8e7134a545f1acc>:0 
  at Xamarin.Forms.Platform.Android.FormsViewGroup.n_OnLayout_ZIIII (System.IntPtr jnienv, System.IntPtr native__this, System.Boolean p0, System.Int32 p1, System.Int32 p2, System.Int32 p3, System.Int32 p4) [0x00008] in <97e56f29d2644ef29e4434660941acc4>:0 
  at (wrapper dynamic-method) System.Object:b1d54703-f2be-40a9-a026-e4aef631fe00 (intptr,intptr,bool,int,int,int,int)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ViewRenderer_2.n_onLayout(Native Method)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ViewRenderer_2.onLayout(ViewRenderer_2.java:47)
	at android.view.View.layout(View.java:16695)
	at android.view.ViewGroup.layout(ViewGroup.java:5328)
	at com.xamarin.forms.platform.android.FormsViewGroup.measureAndLayout(FormsViewGroup.java:29)
	at md5d3faaefeebcb93e0145f431f44dc199e.PopupPageRenderer.n_onLayout(Native Method)
	at md5d3faaefeebcb93e0145f431f44dc199e.PopupPageRenderer.onLayout(PopupPageRenderer.java:48)
	at android.view.View.layout(View.java:16695)
	at android.view.ViewGroup.layout(ViewGroup.java:5328)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:573)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:508)
	at android.view.View.layout(View.java:16695)
	at android.view.ViewGroup.layout(ViewGroup.java:5328)
	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2319)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2032)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1191)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6642)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:777)
	at android.view.Choreographer.doCallbacks(Choreographer.java:590)
	at android.view.Choreographer.doFrame(Choreographer.java:560)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:763)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:145)
	at android.app.ActivityThread.main(ActivityThread.java:5951)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1400)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1195)
```

```
Xamarin caused by: android.runtime.JavaProxyThrowable: System.NullReferenceException: Object reference not set to an instance of an object.
  at Xamarin.Forms.BindableObject.GetContext (Xamarin.Forms.BindableProperty property) [0x00013] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\BindableObject.cs:489 
  at Xamarin.Forms.BindableObject.GetValue (Xamarin.Forms.BindableProperty property) [0x0000e] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\BindableObject.cs:54 
  at Xamarin.Forms.VisualElement.get_Width () [0x00000] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\VisualElement.cs:276 
  at Xamarin.Forms.VisualElement.SetSize (Double width, Double height) [0x00000] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\VisualElement.cs:761 
  at Xamarin.Forms.VisualElement.set_Bounds (Rectangle value) [0x0005d] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\VisualElement.cs:156 
  at Xamarin.Forms.VisualElement.Layout (Rectangle bounds) [0x00000] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\VisualElement.cs:510 
  at Xamarin.Forms.Layout.LayoutChildIntoBoundingRegion (Xamarin.Forms.View child, Rectangle region, SizeRequest childSizeRequest) [0x001c0] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\Layout.cs:275 
  at Xamarin.Forms.StackLayout.LayoutChildren (Double x, Double y, Double width, Double height) [0x00092] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\StackLayout.cs:52 
  at Xamarin.Forms.Layout.UpdateChildrenLayout () [0x000c7] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\Layout.cs:228 
  at Xamarin.Forms.Layout.OnSizeAllocated (Double width, Double height) [0x0000f] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\Layout.cs:193 
  at Xamarin.Forms.VisualElement.SizeAllocated (Double width, Double height) [0x00000] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\VisualElement.cs:628 
  at Xamarin.Forms.Layout+<>c.<OnChildMeasureInvalidated>b__39_0 () [0x00080] in C:\BuildAgent\work\aad494dc9bc9783\Xamarin.Forms.Core\Layout.cs:331 
  at Java.Lang.Thread+RunnableImplementor.Run () [0x0000b] in /Users/builder/data/lanes/3415/7db2aac3/source/monodroid/src/Mono.Android/src/Java.Lang/Thread.cs:36 
  at Java.Lang.IRunnableInvoker.n_Run (IntPtr jnienv, IntPtr native__this) [0x00009] in /Users/builder/data/lanes/3415/7db2aac3/source/monodroid/src/Mono.Android/platforms/android-23/src/generated/Java.Lang.IRunnable.cs:81 
  at (wrapper dynamic-method) System.Object:f5d40293-96f5-48bf-a0c4-ec1587be11e4 (intptr,intptr)
	at mono.java.lang.RunnableImplementor.n_run(Native Method)
	at mono.java.lang.RunnableImplementor.run(RunnableImplementor.java:30)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:135)
	at android.app.ActivityThread.main(ActivityThread.java:5254)
	at java.lang.reflect.Method.invoke(Native Method)
	at java.lang.reflect.Method.invoke(Method.java:372)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:903)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:698)
```

```
Xamarin caused by: android.runtime.JavaProxyThrowable: System.NullReferenceException: Object reference not set to an instance of an object
  at Xamarin.Forms.BindableObject.GetContext (Xamarin.Forms.BindableProperty property) [0x00013] in <b0fc14d4e5b04749b7241d1235a68329>:0 
  at Xamarin.Forms.BindableObject.GetValue (Xamarin.Forms.BindableProperty property) [0x00016] in <b0fc14d4e5b04749b7241d1235a68329>:0 
  at Xamarin.Forms.ListView.get_IsGroupingEnabled () [0x00000] in <b0fc14d4e5b04749b7241d1235a68329>:0 
  at Xamarin.Forms.Platform.Android.ListViewAdapter.get_Count () [0x00013] in <95895bcb439e42ebb8e7134a545f1acc>:0 
  at Android.Widget.BaseAdapter.n_GetCount (System.IntPtr jnienv, System.IntPtr native__this) [0x00009] in <b0aedfc9fdac4d7c9b6250944b7851ac>:0 
  at (wrapper dynamic-method) System.Object:aa5c7199-4af7-4258-8b54-a0117ea0504c (intptr,intptr)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ListViewAdapter.n_getCount(Native Method)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ListViewAdapter.getCount(ListViewAdapter.java:44)
	at android.widget.HeaderViewListAdapter.getCount(HeaderViewListAdapter.java:132)
	at android.widget.ListView.layoutChildren(ListView.java:1599)
	at android.widget.AbsListView.onLayout(AbsListView.java:2201)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at android.support.v4.widget.SwipeRefreshLayout.onLayout(SwipeRefreshLayout.java:596)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ListViewRenderer.n_onLayout(Native Method)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ListViewRenderer.onLayout(ListViewRenderer.java:65)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at com.xamarin.forms.platform.android.FormsViewGroup.measureAndLayout(FormsViewGroup.java:29)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ViewRenderer_2.n_onLayout(Native Method)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.ViewRenderer_2.onLayout(ViewRenderer_2.java:47)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at com.xamarin.forms.platform.android.FormsViewGroup.measureAndLayout(FormsViewGroup.java:29)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.VisualElementRenderer_1.n_onLayout(Native Method)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.VisualElementRenderer_1.onLayout(VisualElementRenderer_1.java:49)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at com.xamarin.forms.platform.android.FormsViewGroup.measureAndLayout(FormsViewGroup.java:29)
	at md5270abb39e60627f0f200893b490a1ade.NavigationPageRenderer.n_onLayout(Native Method)
	at md5270abb39e60627f0f200893b490a1ade.NavigationPageRenderer.onLayout(NavigationPageRenderer.java:63)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at com.xamarin.forms.platform.android.FormsViewGroup.measureAndLayout(FormsViewGroup.java:29)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.PlatformRenderer.n_onLayout(Native Method)
	at md5b60ffeb829f638581ab2bb9b1a7f4f3f.PlatformRenderer.onLayout(PlatformRenderer.java:55)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at android.widget.RelativeLayout.onLayout(RelativeLayout.java:1080)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at android.widget.LinearLayout.setChildFrame(LinearLayout.java:1743)
	at android.widget.LinearLayout.layoutVertical(LinearLayout.java:1586)
	at android.widget.LinearLayout.onLayout(LinearLayout.java:1495)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at android.widget.FrameLayout.layoutChildren(FrameLayout.java:336)
	at android.widget.FrameLayout.onLayout(FrameLayout.java:273)
	at com.android.internal.policy.PhoneWindow$DecorView.onLayout(PhoneWindow.java:2680)
	at android.view.View.layout(View.java:16833)
	at android.view.ViewGroup.layout(ViewGroup.java:5438)
	at android.view.ViewRootImpl.performLayout(ViewRootImpl.java:2474)
	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2131)
	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1233)
	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6552)
	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:919)
	at android.view.Choreographer.doCallbacks(Choreographer.java:710)
	at android.view.Choreographer.doFrame(Choreographer.java:645)
	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:905)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:148)
	at android.app.ActivityThread.main(ActivityThread.java:5582)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```